Load source map only from last directive

zuohaocheng opened the pull request:
<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. -->

**What kind of change does this PR introduce?**
<!-- E.g. a bugfix, feature, refactoring, build related change, etc… -->
bugfix

**Did you add tests for your changes?**
Yes.

**Summary**

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->
There is a library bundled by webpack with `style-loader`. A consumer tries to load it through webpack with `source-map-loader`, but it fails to build.



The root cause is `source-map-loader` incorrectly thinks the string literal `"\n/*# sourceMappingURL=data:application/json;base64,"` is a directive for source map, it tries to find the source map here(while the actual source map directive is in the end of file), and fails.

Per source map spec, the directive should appear at the last line of js file. Thus it makes sense to use last match as the one we read source map from.

**Does this PR introduce a breaking change?**
No.

**Other information**

Relates to: webpack-contrib/style-loader#181

d3viant0ne commented on index.js:
```diff
@@ -22,6 +23,7 @@ module.exports = function(input, inputMap) {
 	var addDependency = this.addDependency;
 	var emitWarning = this.emitWarning || function() {};
 	var match = input.match(regex1) || input.match(regex2);
+	console.log(match);
```
Please remove the debug statement

d3viant0ne commented:
Also, please rebase with current master to pick up the Travis build.

zuohaocheng commented:
@d3viant0ne Thanks for reviewing, I've removed the line and rebased.

There was an issue on test: The line ends on Windows is `CRLF` by `git` default setting, so the assertion couldn't pass. I added a step to remove all `CR`s in reading fixtures, making tests could be passed with `CRLF`.

d3viant0ne commented:
Just gave it a quick triage, we will give this a good look in the morning. I've linked the two issues and we will be sure to coordinate this with the associated PR in style-loader.

ekulabuhov commented on test/index.test.js:
```diff
@@ -33,7 +33,7 @@ function execLoader(filename, callback) {
 		}
 	};
-	var res = loader.call(context, fs.readFileSync(filename, "utf-8"));
+	var res = loader.call(context, fs.readFileSync(filename, "utf-8").replace(/\r/g, ''));
```
Split in two lines: (e.g. `var res = ... <newline> res = res.replace(/\r/g, '')` and add a comment explaining the addition.

ekulabuhov commented:
> Per source map spec, the directive should appear at the last line of js file
Would it work if you only checked the last line? If I understand correctly current regex is checking the whole file.

Great PR 👍

zuohaocheng commented:
@ekulabuhov Thanks and updated accordingly.

I'm afraid that some non-standard source inputs would be broken, if checking only the last line for source map directive.

michael-ciniawsky commented:
@d3viant0ne :shipit: ?
